File: android/src/main/java/com/ospedrobilro/capacitormanifestplaceholder/ManifestPlaceholderCapPlugin.java
package com.ospedrobilro.capacitormanifestplaceholder;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ManifestPlaceholderCap")
public class ManifestPlaceholderCapPlugin extends Plugin {

    private ManifestPlaceholderCap implementation = new ManifestPlaceholderCap();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}

File: ios/Sources/ManifestPlaceholderCapPlugin/ManifestPlaceholderCapPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ManifestPlaceholderCapPlugin)
public class ManifestPlaceholderCapPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ManifestPlaceholderCapPlugin"
    public let jsName = "ManifestPlaceholderCap"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = ManifestPlaceholderCap()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
